source/ini_file.cpp

```diff
@@ -119,7 +119,7 @@ bool ini_file::save()
 
 	std::error_code ec;
 	const std::filesystem::file_time_type modified_at = std::filesystem::last_write_time(_path, ec);
-	if (!ec && modified_at > _modified_at)
+	if (!ec && (modified_at - _modified_at) > std::chrono::seconds(2))
```

Comment:
Could you explain what prompted this change?

Reply:
See my 'screenshot' add-on https://github.com/cot6/reshade-addons/blob/c1c7c6cb1f7b94c6491ce226c69c7f467b51ff37/src/addon-screenshot/dllmain.cpp#L126-L140

This is a special behavior to avoid that effects bundled with this add-on are not loaded if 'Load only enabled effects' is enabled.

This add-on does load and modify the preset's contents before the host loading the effects.
However, the host has a cache of INI files, and if it detects changes written by another of these, the save will fail because of the condition preferring the latest changes.
The reason why 2 seconds rather than 1, because FAT32 can handle the last write time with 2-second precision.
And only having a cache of 0.5 seconds is too short.
This allows the add-on have an INI cache, and the host can still handle own successfully.

Reply:
However, reviewing the code now, it appears to be dependent on Windows timing, it does not close the INI file before try to reload (set dummy preprocessor definition).
When I looked at it in Procmon while working on this issue, if it does not close the file byself, the syscall is called by System process, not the EXE.